custom-metrics-stackdriver-adapter/pkg/adapter/provider/provider.go

```diff
@@ -55,7 +56,7 @@ func (c realClock) Now() time.Time {
 
 // StackdriverProvider is a provider of custom metrics from Stackdriver.
 type StackdriverProvider struct {
-	apiserverClient    rest.Interface
+	kubeClient         rest.Interface
```

Comment:
BTW is there any reason why this is rest.Interface rathern than [CoreV1Client](https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/core_client.go#L47)? The latter offers you native pods methods (see [here](https://github.com/kubernetes/client-go/blob/master/kubernetes/typed/core/v1/pod.go#L35) and you can avoid passing `runtime.Object` and conversion problems below.

Reply:
As discussed offline, I agree it's better to switch to CoreV1Client.